packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js

```diff
@@ -116,41 +115,37 @@ export default function useNestedSettingsUpdate(
 			newSettings.__experimentalDirectInsert = __experimentalDirectInsert;
 		}
 
-		if ( ! isShallowEqual( blockListSettings, newSettings ) ) {
```

Comment:
This change was guided not really by measurable performance concerns, but by the fact that the rerender triggered by a state update in a microtask leads to `act()` warnings. At the time when the rerender happens, there is no `act()` call active to catch it.

There are reasons to believe that an extra execution of the update loop is less costly that a React rerender. Because less code is executed. The worst thing that can happen is that they are both the same performance-wise.

Reply:
Definitely understand the guidance or rationale. 

I suppose my point is that I found it interesting you modified the application code for a test. I generally avoid doing that. So, I likely would not have considered refactoring this code for a test, but instead leveraged additional `act` calls to resolve the warning. That is why I expressed what I perceived to be _potential_ negatives from the changes and sought your thoughts on those. 

With that clarification, this still seems like the best approach. 👍🏻

Reply:
> I suppose my point is that I found it interesting you modified the application code for a test.

In this case I think the change is a good idea on its own, even without the affected test. The code used a pattern like this:
```js
const settings = useSelect( s => s( editor ).getSettings() );
const { setSetting } = useDispatch( editor );
function handleChange( key, value ) {
  if ( settings[ key ] !== value ) {
    setSetting( key, value );
  }
}
```
Here a React component is selecting the `settings` value, although it doesn't use it for render at all. Any change in `settings` will trigger a rerender that's guaranteed to be a noop. The value is used only to detect if the value is really changing.

Why not put the value change check directly into `setSetting`? The action and it's associated reducer has all the information it needs to do the check. And it avoids the rerender.
```js
const { setSetting } = useDispatch( editor );
function handleChange( key, value ) {
  setSetting( key, value );
}
```

Reply:
That makes sense. Thank you for elaborating.